### ISSUE TITLE
Media runtime sometimes not shown (shown as `0m 0s`)

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#40234]
### Description

Media runtime sometimes not shown (shown as `0m 0s`)

Found while testing https://github.com/brave/brave-browser/issues/32585

### Steps to reproduce

1. install `1.70.41`
2. launch Brave
3. load `https://www.youtube.com/watch?v=4iNMyilmDfg`
4. restart Brave (to pick up Griffin seeds)
5. load `https://www.youtube.com/watch?v=4iNMyilmDfg` again
6. click on the music note icon to `Add to Playlist`
7. 

### Actual result

Runtime is shown as `0m 0s`

<img width="1316" alt="Screen Shot 2024-08-02 at 2 12 01 PM" src="https://github.com/user-attachments/assets/b135116c-90a7-4647-86db-cfb8319924e4">

### Expected result

Should show the media's ` hours` ` minutes` ` seconds` runtime

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

```
Brave | 1.70.41 Chromium: 127.0.6533.88 (Official Build) nightly (x86_64)
-- | --
Revision | 9303bae9fe07bc2ec7c0cdef139e87304b2d6c96
OS | macOS Version 11.7.10 (Build 20G1427)
```

### Channel information

- [ ] release (stable)
- [X] beta
- [X] nightly

### Reproducibility

- [X] with Brave Shields disabled
- [X] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

@aguscruiz @sangwoo108 @szilardszaloki @rebron @brave/qa-team

**Comment by MadhaviSeelam** [comment]
Reproduced the issue with twitch.tv clips
https://www.twitch.tv/mrskilted/clip/ExcitedDistinctGarageANELE-HE82zvRFIxiX8Xkj

<img width="229" alt="image" src="https://github.com/user-attachments/assets/51e5234c-1989-4e57-9b49-e7aea9a699c0">

**Comment by sangwoo108** [comment]
Yeah, sometimes we can't get the duration properly at the point of caching. I was thinking of updating it when the video is played.

**Comment by szilardszaloki** [comment]
Hey all,
Just a heads-up that we won't have to deal with this once we're on the MediaSession API.
